### PR TITLE
header requestparser added

### PR DESCRIPTION
**Pull request by younger15**

### CODE REVIEW

File: src/Header/RequestParser.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "./OperateData.h"
```
Comment (john0312) [inline src/Header/RequestParser.h]: "./" in include is not necessary. If in doubt, set the include path in CMake.

File: src/Header/RequestParser.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    static OperateData ParseHeader(std::string request);
+    //give request string return body / data
+    static OperateData ParseData(std::string request);
+private:
+    //singleton code convention method.
+    RequestParser();
+};
+
+namespace{
+    RequestParser* onlyRequestParser;
```
Comment (john0312) [inline src/Header/RequestParser.h]: This should go in the .cc file.

Comment (john0312) [reply src/Header/RequestParser.h]: Please add comment to note that it is the singleton instance.

Comment (john0312) [reply src/Header/RequestParser.h]: Note: It is recommended that you study why putting this in the header is a problem. If in doubt, ask me in class.

File: src/Header/RequestParser.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#include "./OperateData.h"
+
+//used to make request string to OperateData(JSON like) format.
+class RequestParser
+{
+public:
+    //singleton code convention method.
+    static RequestParser* Get();
+    //give request string return header
+    static OperateData ParseHeader(std::string request);
```
Comment (john0312) [inline src/Header/RequestParser.h]: Consider returning a unique_ptr of OperateData to avoid copying.

File: src/Header/RequestParser.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "./OperateData.h"
+
+//used to make request string to OperateData(JSON like) format.
```
Comment (john0312) [inline src/Header/RequestParser.h:11]: Please add a very simple usage example in the comment. i.e.
```
std::string request = "GET / HTTP/1.1\n...";
unique_ptr<OperateData> data = RequestParser::ParseHeader(request);
// data.method() now contains the method, data.headers() now contains the headers... etc
```

File: src/Header/RequestParser.h (hunk 3 above)
Comment (john0312) [inline src/Header/RequestParser.h]: Please use const reference for input.

File: src/Header/RequestParser.h (hunk 1 above)
Comment (younger15) [reply src/Header/RequestParser.h]: Include path changed

File: src/Header/RequestParser.h (hunk 2 above)
Comment (younger15) [reply src/Header/RequestParser.h]: Namespace removed

File: src/Header/RequestParser.h (hunk 4 above)
Comment (younger15) [reply src/Header/RequestParser.h:11]: Example added

File: src/Header/RequestParser.h (hunk 3 above)
Comment (younger15) [reply src/Header/RequestParser.h]: Have changed to return unique_ptr.

File: src/Header/RequestParser.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+#include "OperateData.h"
+
+//used to make request string to OperateData(JSON like) format.
+//string inRequest "GET /?token=xxx HTTP/1.1/n..." => unique_ptr<OperateData> data = ParseHeader(inRequest)
+//data will be a json like format map, header and body will be seperate to two OperateData
+class RequestParser
+{
+public:
+    //singleton code convention method.
```
Comment (john0312) [inline src/Header/RequestParser.h:18]: If all methods are static, you don't need singleton. Simply use the static methods directly.

**Comment by john0312** [comment]
Hello @mike820324 (Ruei-Min Jiang), please take a look at this commit/PR. Thank you.

File: src/Header/RequestParser.h (hunk 3 above)
Comment (younger15) [reply src/Header/RequestParser.h]: Done

File: src/Header/RequestParser.h (hunk 5 above)
Comment (younger15) [reply src/Header/RequestParser.h:18]: Parse function changed to non static

**Comment by john0312** [comment]
LGTM.

**Pull request merged**